planus/src/errors.rs

```diff
@@ -17,16 +20,39 @@ impl std::error::Error for Error {
     }
 }
 
+/// The possible errors in planus when reading data from a serialized buffer.
 #[derive(Copy, Clone, Debug)]
 #[non_exhaustive]
 pub enum ErrorKind {
+    /// The offset was out of bounds.
     InvalidOffset,
+    /// The buffer was too short while validating a length field.
     InvalidLength,
-    UnknownEnumTag { source: UnknownEnumTagKind },
-    UnknownUnionTag { tag: u8 },
-    InvalidVtableLength { length: u16 },
-    InvalidUtf8 { source: core::str::Utf8Error },
+    /// An enum contained an unknown value. For forward compatibility this
+    /// error should be handled appropriately.
+    UnknownEnumTag {
+        /// The enum value that wasn't recognized.
+        source: UnknownEnumTagKind,
+    },
+    /// An union contained an unknown variant. For forward compatibility this
+    /// error should be handled appropriately.
+    UnknownUnionTag {
+        /// The union tag that wasn't recognized.
+        tag: u8,
+    },
+    /// A vtable was too large for the remaining buffer.
```

Comment:
Not only too large, also of wrong sizes like uneven or with a size that cuts off half of a union field.